### PR TITLE
Release bump systemd-boot to use new keys

### PR DESCRIPTION
**Pull request by dmcilvaney**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

`systemd-boot` also requires the new secureboot keys, version bump so we re-sign them.

Add space recommends/conflicts as https://github.com/microsoft/azurelinux/pull/10995/files#diff-9dd7fef1e850ad265fea94cbe5a86e163172ee833f8ec954d946d57b0ca44ce4

###### Change Log  

- Version bump `systemd-boot-signed` and `systemd` to force a resign with the new keys

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- 575013955

###### Test Methodology

- tbd, will need testing before final publish

### CODE REVIEW

**Comment by dmcilvaney** [comment]
Test pipelines have failures, but at least systemd looks expected

File: SPECS-SIGNED/systemd-boot-signed/systemd-boot-signed.spec [hunk 1]
```diff
@@ -90,6 +93,9 @@ popd
 /boot/efi/EFI/BOOT/grubx64.efi
 
 %changelog
+* Thu Dec 12 2024 Daniel McIlvaney <damcilva@microsoft.com> - 255-19
+- Version bump to force signing with new Azure Linux secure boot key
```
Comment (christopherco) [inline SPECS-SIGNED/systemd-boot-signed/systemd-boot-signed.spec:97]: You're also adding package Recommends & Conflicts to ensure proper upgrade behavior. Please add a changelog note about this.

**Review by christopherco — COMMENTED** [review]
Aside from the 1 comment, changes look good

File: SPECS-SIGNED/systemd-boot-signed/systemd-boot-signed.spec (hunk 1 above)
Comment (dmcilvaney) [reply SPECS-SIGNED/systemd-boot-signed/systemd-boot-signed.spec:97]:
```suggestion
- Version bump to force signing with new Azure Linux secure boot key
- Add confilcts/recommends on shim to ensure the keys match
```

**Comment by dmcilvaney** [comment]
Per discussion, will target 3.0-dev, followed by cherry-pick to 3.0. Fasttrack may not be fully ready for this type of release.

**Review by jslobodzian — APPROVED** [review]

**Pull request merged**